src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Component, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import { ChartType, ChartOptions } from 'chart.js';
+import { } from 'ng2-charts';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+@Component({
+	selector: 'chart-component',
+	templateUrl: decodeURI(require.toUrl('./chart.component.html'))
+})
+export class Chart extends Disposable {
+
+	private _labels: string[] = [];
+	private _data: number[] = [];
+	private _colors: string[] = [];
+	public doughnutChartColors: any[] = [];
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef
+	) {
+		super();
+	}
+
+	public set labels(val: string[]) {
+		this._labels = val;
+		this._changeRef.detectChanges();
+	}
+
+	public get labels(): string[] {
+		return this._labels;
+	}
+
+	public set data(val: number[]) {
+		this._data = val;
+		this._changeRef.detectChanges();
+	}
+
+	public get data(): number[] {
+		return this._data;
+	}
+
+	public get colors(): string[] {
+		return this._colors;
+	}
+
+	public set colors(val: string[]) {
+		this._colors = val;
+		this.doughnutChartColors = [
+			{
+				backgroundColor: this._colors
+			}
+		];
+		this._changeRef.detectChanges();
+	}
+
+	doughnutChartType: ChartType = 'doughnut';
+
+	public doughnutChartOptions: ChartOptions = {
+		responsive: true,
+		maintainAspectRatio: false
+	};
+
+	public chartClicked(e: any): void {
```

Comment:
Keyboard navigation can be added after ChartJS vBump

Reply:
@kburtram I think we should follow what we did for dashboard: when the setting `editor.accessibilitySupport` is enabled, we should show the data in table format so that it is accessible.  https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/dashboard/browser/widgets/insights/insightsWidget.component.ts#L310

what do you think?

Reply:
@alanrenmsft I agree that if we can't provide directly accessible charts the UI should fall back to a tabular view of the equivalent data.  Though not sure if this should be forced at the charting API layer or controlled at the extension layer.  For example, the extension may prefer to make the UI accessible through a different mechanism.  

Probably makes sense for the default behavior to be that the chart falls back to a table, and this can be overridden by parameter.  I don't think we can add the API where the default behavior doesn't meet accessibility requirements as this will come to us as a bug as soon as any extension takes a dependency on it.

Reply:
Thanks Karl, @deepaksa1 , let's introduce an option `showAsTableInAccessibleMode` and the default value is true.

Reply:
Is there a reason we would want to let users override this? I think having it always fall back to a table in accessible mode would be safer - extensions can always check the config to see if accessible mode is enabled and use another component if it is if they really want.